docs/syntax/naming.md

```diff
@@ -44,11 +44,10 @@ following rules around naming:
   - Each 'word' in the identifier must be lower-case or a number.
   - Words in the identifier are separated using `_`.
   - Numbers may not occur as the first 'word' in an identifier.
-- An identifier named as above can be referred to by capitalizing the first
-  letter of each 'word' in the identifier. This is known as 'referent' form.
 - No mixed-format names are allowed (e.g. `HTTP`, `foO`, `make_New`, or
   `Make_new`). These should be rejected by the compiler.
```

Comment:
This isn't true any more so we should remove (we allow `AWS_Profile` and `SQLite` for example).